src/storage/store/rel_table.cpp

```diff
@@ -166,7 +168,30 @@ bool RelTable::scanInternal(Transaction* transaction, TableScanState& scanState)
     return scanState.scanNext(transaction);
 }
 
+void RelTable::checkRelMultiplicityConstraint(Transaction* transaction,
+    const TableInsertState& state) const {
+    const auto& insertState = state.constCast<RelTableInsertState>();
+    KU_ASSERT(insertState.srcNodeIDVector.state->getSelVector().getSelSize() == 1 &&
+              insertState.dstNodeIDVector.state->getSelVector().getSelSize() == 1);
+
+    const auto throwFunc = [](const std::string& tableName, offset_t nodeOffset,
+                               RelDataDirection direction) {
+        throw RuntimeException(ExceptionMessage::violateRelMultiplicityConstraint(tableName,
+            std::to_string(nodeOffset), RelDataDirectionUtils::relDirectionToString(direction)));
+    };
```

Comment:
I wonder how L177-L181 differs from defining a static function and passed as lambda into `throwIfNodeHasRels`. Will that trigger copying?

Reply:
In this case I think there's a good chance that copying will trigger. It's possible to avoid the copy while keeping the lambda but it's probably best to keep it simple and make it a static function.

Reply:
updated